mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/BindingContext.java

```diff
@@ -16,4 +19,18 @@
 	 */
 	BeanResolver beanResolver();
 
+	/**
+	 * @param name The name of the param
+	 * @return Get a param defined for the binder by the given name
+	 * @throws SearchException if it does not exist a param having such name
+	 */
+	Object param(String name);
+
+	/**
+	 * @param name The name of the param
+	 * @return Get an optional param defined for the binder by the given name,
+	 * a param having such name may either exist or not.
+	 */
+	Optional<Object> optionalParam(String name);
```

Comment:
I still think `paramOptional` would be better for discovery: as a developper, you start typing `param`, and the IDE will suggest both `param()` and `paramOptional()`. Whereas with this name, a lot IDEs will only suggest `param()`.

It's also not as "grammatically incorrect" as you would think. You can think of "paramOptional" as "param[eter] [as an] Optional".